Could not link fiber 1.84.0 as shared library on Windows 10

Hello! We are trying to package the Boost 1.84.0 on Conan but we are facing a linkage error related to the new `boost::fibers::make_stack_allocator_wrapper` introduced on [1.84.0](https://github.com/boostorg/fiber/compare/boost-1.83.0...boost-1.84.0)

We have the follow error when trying to link to the fiber example [work_sharing.cpp](https://github.com/boostorg/fiber/blob/develop/examples/work_sharing.cpp):

```
boost/1.84.0 (test package): RUN: cmake --build "C:\Users\uilia\Development\conan\conan-center-index\recipes\boost\all\test_package\build\msvc-193-x86_64-17-release" --config Release                                                                                                                                        
MSBuild version 17.3.1+2badb37d1 for .NET Framework                                                                                                                                                                                                                                                                           
  1>Checking Build System                                                                                                                                                                                                                                                                                                                                                                                                                                                 
  Building Custom Rule C:/Users/uilia/Development/conan/conan-center-index/recipes/boost/all/test_package/CMakeLists.txt                                                                                                                                                                                                      
  fiber.cpp                                                                                                                                                                                                                                                                                                                   
C:/Users/uilia/.conan2/p/b/boost9b0969d51319d/p/include\boost/fiber/stack_allocator_wrapper.hpp(85,43): error C2491: 'boost::fibers::make_stack_allocator_wrapper': definition of dllimport function not allowed [C:\Users\uilia\Development\conan\conan-center-index\recipes\boost\all\test_package\build\msvc-193-x86_64-17 
-release\fiber_exe.vcxproj]                                                                                                                                                                                                                                                                                                   
  Building Custom Rule C:/Users/uilia/Development/conan/conan-center-index/recipes/boost/all/test_package/CMakeLists.txt                                                                                                                                                                                                      
```

In summary, we using the exactly same configuration for Boost 1.83.0 and it works perfectly with the same example file in this repository. 

All build and test logs are available on https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/21747/11-windows-visual_studio/boost/1.84.0//summary.json

We use b2 to build Boost, including fiber. The CMake is only when consuming Boost, so we are not trying unofficial support.

Some extra information about the generated fiber.dll:

![Screenshot (93)](https://github.com/boostorg/fiber/assets/4870173/4ed02813-3a25-4368-a09c-5ecfb7485f9c)


#### Environment

Boost version: 1.84.0
OS: Windows 10
Compiler: MSVC 193
Build Type: Release
Runtime: Dynamic (MD)

#### Steps to reproduce

```
git clone -b boost-1.84.0 https://github.com/toge/conan-center-index.git
cd recipes/boost/all
conan create . --version=1.84.0
```

OR, just build Boost as usual and try to link to work_sharing.cpp.

--- a/include/boost/fiber/stack_allocator_wrapper.hpp
+++ b/include/boost/fiber/stack_allocator_wrapper.hpp
@@ -82,7 +82,7 @@ private:
 };
 
 template <typename StackAllocator, typename ... Args>
-BOOST_FIBERS_DECL stack_allocator_wrapper make_stack_allocator_wrapper(Args && ... args)
+stack_allocator_wrapper make_stack_allocator_wrapper(Args && ... args)
 {
     return stack_allocator_wrapper(
             std::unique_ptr<detail::polymorphic_stack_allocator_base>(

